File: Csharp/CodeWars/src/Resolutions/JumpingNumberSpecialNumbersSeries4.cs
namespace CodeWars.Resolutions;

public class JumpingNumberSpecialNumbersSeries4
{
    public static string JumpingNumber(int number) =>
        Enumerable
        .Range(1, number.ToString().Length - 1)
        .All(x => Math.Abs(number.ToString()[x] - number.ToString()[x - 1]) == 1)
        ? "Jumping!!"
        : "Not!!";
}

File: TypeScript/Codewars/src/resolutions/Jumping-Number-Special-Numbers-Series-4.ts
export function jumpingNumber(n: number) {
	if (String(n).length === 1) return "Jumping!!";

	const result = String(n)
		.split("")
		.map(Number)
		.every((value, index, array) => {
			if (index === array.length - 1) return true;
			return Math.abs(value - array[index + 1]) === 1;
		});

	return result ? "Jumping!!" : "Not!!";
}
